File: python/src/training_data.py
import os
import datetime
import sys
sys.path.append('./src/lib')
from format import __to_json__

class TrainingData:
    def __init__(self, training_data):
        self.training_data = training_data

    def export(self, dirname):
        with open(self.__filename__(dirname), 'w') as f:
            f.write(__to_json__(self.training_data))

    # private

    def __filename__(self, dirname):
        return os.path.join(dirname, 'training_data_{0:%Y%m%d%H%M%S}.json'.format(datetime.datetime.now()))

File: ruby/src/training_data.rb
# rbs_inline: enabled

require_relative './lib/format'

class TrainingData
  include ::Lib::Format

  # @rbs training_data: String
  # @rbs return: void
  def initialize(training_data)
    @training_data = training_data
  end

  # @rbs dirname: String
  # @rbs return: void
  def export(dirname)
    File.write(filename(dirname), to_json(training_data))
  end

  private

  attr_reader :training_data

  # @rbs dirname: String
  # @rbs return: String
  def filename(dirname)
    File.join(dirname, "training_data_#{Time.now.strftime('%F%T').gsub(/[:\-]/, '')}.json")
  end
end
